crates/e2e/macro/src/codegen.rs

```diff
@@ -190,3 +163,49 @@ impl InkE2ETest {
         }
     }
 }
+
+/// Builds the contract at `manifest_path`, returns the path to the contract
+/// bundle build artifact.
+fn build_contract(manifest_path: &str) -> String {
+    use std::process::{
+        Command,
+        Stdio,
+    };
+    let output = Command::new("cargo")
+        .args([
+            "+stable",
+            "contract",
+            "build",
```

Comment:
Since we control `cargo-contract` it might be really nice to have a lib to allow invocation of contract builds...would certainly simplify the code here.

Made a quick start on it here https://github.com/paritytech/cargo-contract/pull/787. 

If you agree it is a good idea I can complete that.